lib/auth/auth_with_roles.go

```diff
@@ -714,6 +714,40 @@ NextNode:
 	return filteredNodes, nil
 }
 
+func (a *ServerWithRoles) checkAccessToNode(server types.Server) error {
```

Comment:
Seems like this function is very similar to `filterNodes`. Any way to reuse one or the other?

Reply:
Updated the `filterNodes` to reuse `checkAccessToNode`.